samza-core/src/test/java/org/apache/samza/metrics/reporter/TestMetricsSnapshotReporterFactory.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.metrics.reporter;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.samza.config.Config;
+import org.apache.samza.config.MapConfig;
+import org.apache.samza.metrics.MetricsRegistry;
+import org.apache.samza.metrics.MetricsRegistryMap;
+import org.apache.samza.serializers.MetricsSnapshotSerdeV2;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.serializers.SerdeFactory;
+import org.apache.samza.system.SystemAdmin;
+import org.apache.samza.system.SystemConsumer;
+import org.apache.samza.system.SystemFactory;
+import org.apache.samza.system.SystemProducer;
+import org.apache.samza.system.SystemStream;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+
+public class TestMetricsSnapshotReporterFactory {
+  private static final String REPORTER = "metrics-reporter";
+  private static final SystemProducer SYSTEM_PRODUCER = mock(SystemProducer.class);
+  private static final Serde<MetricsSnapshot> SERDE = mock(Serde.class);
+
+  private MetricsSnapshotReporterFactory factory;
+
+  @Before
+  public void setup() {
+    this.factory = new MetricsSnapshotReporterFactory();
+  }
+
+  @Test
+  public void testGetProducer() {
+    Config config = new MapConfig(
+        ImmutableMap.of("metrics.reporter.metrics-reporter.stream", "system0.stream0", "systems.system0.samza.factory",
+            MockSystemFactory.class.getName()));
+    assertEquals(SYSTEM_PRODUCER, this.factory.getProducer(REPORTER, config, new MetricsRegistryMap()));
+  }
+
```

Comment:
Would it make sense to add a test to compare the serialized format of a sample object 
before and after this change? 
Similarly for deserialization -- to ensure the new code can deserialize things serialized using previous code.